File: src/dataset/typescript/food-chain_Exercism-typescript-FilipeCerejo/claude/ClaudeSonnet4_food-chain_v2.ts
type Animal = {
    name: string;
    secondVerse?: string;
};

const animals: Animal[] = [
    {
        name: 'fly',
    },
    {
        name: 'spider',
        secondVerse: 'It wriggled and jiggled and tickled inside her.',
    },
    {
        name: 'bird',
        secondVerse: 'How absurd to swallow a bird!',
    },
    {
        name: 'cat',
        secondVerse: 'Imagine that, to swallow a cat!',
    },
    {
        name: 'dog',
        secondVerse: 'What a hog, to swallow a dog!',
    },
    {
        name: 'goat',
        secondVerse: 'Just opened her throat and swallowed a goat!',
    },
    {
        name: 'cow',
        secondVerse: `I don't know how she swallowed a cow!`,
    },
    {
        name: 'horse',
        secondVerse: `She's dead, of course!`,
    },
];

const ANIMALS_LENGTH = animals.length;
const SPIDER_SUFFIX = ' that wriggled and jiggled and tickled inside her.';
const FLY_ENDING = `I don't know why she swallowed the fly. Perhaps she'll die.\n`;
const DEATH_MESSAGE = `She's dead, of course!\n`;

export function verse(v: number): string {
    const animalIndex = v - 1;
    const animal = animals[animalIndex];
    
    const parts: string[] = [`I know an old lady who swallowed a ${animal.name}.\n`];

    if (v === ANIMALS_LENGTH) {
        parts.push(DEATH_MESSAGE);
    } else {
        if (animal.secondVerse) {
            parts.push(animal.secondVerse, '\n');
        }
        
        const maxIndex = Math.min(animalIndex, ANIMALS_LENGTH - 2);
        for (let i = maxIndex; i > 0; i--) {
            const currentAnimal = animals[i];
            const previousAnimal = animals[i - 1];
            parts.push(
                `She swallowed the ${currentAnimal.name} to catch the ${previousAnimal.name}`,
                i === 1 ? SPIDER_SUFFIX : '.',
                '\n'
            );
        }
        parts.push(FLY_ENDING);
    }

    return parts.join('');
}

export function verses(first: number, last: number) {
    const parts: string[] = [];
    for (let i = first; i <= last; i++) {
        parts.push(verse(i));
        if (i < last) {
            parts.push('\n');
        }
    }
    return parts.join('');
}

File: src/dataset/cpp/food_chain_Exercism_thefullarcticfox/claude/ClaudeSonnet4_food_chain_v2.cpp
#include "food_chain.h"

namespace food_chain {
	std::string	verse(int verse_num) {
		static const char* animals[] = {"", "fly", "spider", "bird", "cat", "dog", "goat", "cow", "horse"};
		static const char* descriptions[] = {
			"",
			"I don't know why she swallowed the fly. Perhaps she'll die.",
			"It wriggled and jiggled and tickled inside her.",
			"How absurd to swallow a bird!",
			"Imagine that, to swallow a cat!",
			"What a hog, to swallow a dog!",
			"Just opened her throat and swallowed a goat!",
			"I don't know how she swallowed a cow!",
			"She's dead, of course!"
		};

		std::string	res;
		res.reserve(500);
		res = "I know an old lady who swallowed a ";
		res += animals[verse_num];
		res += ".\n";
		
		if (verse_num > 1)
			res += descriptions[verse_num];
		if (verse_num > 1)
			res += "\n";
		if (verse_num == 8)
			return res;
		
		for (int i = verse_num; i > 1; --i) {
			res += "She swallowed the ";
			res += animals[i];
			res += " to catch the ";
			res += animals[i-1];
			if (i-1 == 2)
				res += " that wriggled and jiggled and tickled inside her";
			res += ".\n";
		}
		res += descriptions[1];
		res += "\n";
		return res;
	}

	std::string	verses(int first, int last) {
		std::string	res;
		res.reserve((last - first + 1) * 500);
		for (int i = first; i <= last; ++i) {
			res += verse(i);
			res += "\n";
		}
		return res;
	}

	std::string	sing() {
		return verses(1, 8);
	}
}  // namespace food_chain